observer.py

```diff
@@ -65,16 +66,21 @@ async def main(args):
                 asyncio.sleep(0.4)
                 continue
 
+            crypto_products = set([p.attrs["base"] for p in products if p.attrs["asset_type"] == 'Crypto'])
+            coingecko_prices = get_coingecko_prices(crypto_products)
```

Comment:
i think this price update should run in a separate asyncio loop and cache the results in a variable. this loop can then read the most recent prices from that variable. You definitely don't want to couple together the 0.4s alerting times with the 30s coingecko rate limit, nor do you want to block the alerts while waiting for the coingecko api to return.